File: project3/person.cc
#include "person.h"

Person::Person(const string name, const int floor, const int classroom){
    // cout << "A person with name: " << name << " has been created" << endl;
    this->name = name;
    this->floor_id = floor;
    this->class_id = classroom;
    this->classroom = false;
    this->tiredness = 0;
    this->position = OUT_OF_SCHOOL;
}

Person::~Person(){
    // cout << "A person with name " << this->name << " is about to be destroyed" << endl;
}

string Person::get_name(){
    return this->name;
}

int Person::get_class_id(){
    return this->class_id;
}

int Person::get_floor_id(){
    return this->floor_id;
}

void Person::set_classroom(const bool pos){
    this->classroom = pos;
}

void Person::set_position(const int pos){
    this->position = pos;
}

int Person::get_position(void){
    return this->position;
}

void Person::add_tiredness(const int i){
    this->tiredness += i;
}

void Person::print(void){
    cout << this->name << ", " << this->tiredness << endl;
}

int Person::get_tiredness(void){
    return this->tiredness;
}

Student::Student(const string name_init, const int floor, const int classroom):Person(name_init, floor, classroom){
    cout << "A new student with name: " << name_init << " ,class id: " << classroom << " and floor id: " << floor  <<" has been created" << endl;
}

Student::~Student(){
    cout << "A student with name " << Person::get_name() << " is about to be destroyed" << endl;
}

void Student::print(void){
    Person::print();
}

Teacher::Teacher(const string name_init, const int floor, const int classroom):Person(name_init, floor, classroom){
    if(name_init != "Nothing")
        cout << "A new teacher with name: " << name_init << " ,class id: " << classroom << " and floor id: " << floor  <<" has been created" << endl;
}

Teacher::~Teacher(){
    if(Person::get_name() != "Nothing")
        cout << "A teacher with name: " << Person::get_name() << " is about to be destroyed" << endl;
}

void Teacher::print(void){
    cout << "Teacher: "<< endl;
    Person::print();
}

void Teacher::place(void){
    Person::set_classroom(true);
}

File: java/Person.java
public class Person{
    String name;
    int class_id;
    int floor_id;
    boolean classroom;
    int tiredness;
    int position;

    public Person(String name_init, int floor, int classroom_id){
        name = name_init;
        class_id = classroom_id;
        floor_id = floor;
        classroom = false;
        tiredness = 0;
        position = Globals.OUT_OF_SCHOOL;
        // System.out.println("A new person has been created with name: " + name);    
    }

    public String toString(){
        return String.format(name + ", " + tiredness);
    }

    public String get_name(){
        return name;
    }

    public int get_class_id(){
        return class_id;
    }

    public int get_floor_id(){
        return floor_id;
    }

    public void set_classroom(boolean pos){
        classroom = pos;
    }

    public void add_tiredness(int i){
        tiredness = tiredness + i;
    }

    public int get_tiredhess(){
        return tiredness;
    }

    public void print(){
        System.out.println(name + " "  + tiredness);
    }

    public void set_position(int pos){
        position = pos;
    }

    public int get_position(){
        return position;
    }

}

class Student extends Person{
    public Student(String name_init, int floor_init, int class_init){
        super(name_init, floor_init, class_init);
        if(name != "Nothing")
            System.out.println("A new student has been created with name: " + name + ", class id: " + class_id + ", floor id: " + floor_id);    
    }
    
    public void print(){
        super.print();
    }
}

class Teacher extends Person{
    public Teacher(String name_init, int floor_init, int class_init){
        super(name_init, floor_init, class_init);
        if(name != "Nothing")
            System.out.println("A new teacher has been created with name: " + name + ", class id: " + class_id + ", floor id: " + floor_id); 
    }

    public void print(){
        super.print();
    }

    public void place(){
        super.set_classroom(true);
    }
}






